mycompany-seeq-link-connector-myconnector/src/main/java/com/mycompany/seeq/link/connector/MyConnection.java

```diff
@@ -142,87 +164,94 @@ public void disconnect() {
+        // root
+        // asset in the Seeq database.
+        String rootAssetId = this.syncRootAsset();
+
         // Do whatever is necessary to generate the list of signals you want to show up in Seeq. It is generally
         // preferable to use a "streaming" method of iterating through the tags. I.e., try not to hold them all in
         // memory because it is harder to scale to indexing hundreds of thousands of signals. Although these examples
         // use Iterators (which can also be composed in a 'lazy' manner), you may want to consider using Java 8's
         // Streams, which are often friendlier and more convenient to use.
 
-        // Loop through all of the tags in our simulated datasource and tell Seeq Server about them
-        Iterator<DatasourceSimulator.Tag> tags = this.datasourceSimulator.getTags();
-        while (tags.hasNext()) {
-            DatasourceSimulator.Tag tag = tags.next();
-            SignalWithIdInputV1 signal = new SignalWithIdInputV1();
-
-            // The Data ID is a string that is unique within the data source, and is used by Seeq when referring
-            // to signal / condition / asset data. Data ID is a string and does not need to be numeric, even
-            // though we are just using a number in this example.
-            signal.setDataId(String.format("%d", tag.getId()));
-
-            // The Name is a string that is displayed in the UI. It can change (typically as a result of a
-            // rename operation happening in the source system), but the unique Data ID preserves appropriate
-            // linkages.
-            signal.setName(tag.getName());
-
-            // The interpolation method is the final piece of critical information for a signal.
-            signal.setInterpolationMethod(tag.getStepped()
-                    ? DatasourceConnectionServiceV2.InterpolationMethod.Step
-                    : DatasourceConnectionServiceV2.InterpolationMethod.Linear);
-
-            // putSignal() queues items up for performance reasons and writes them in batch to the server.
-            //
-            // If you need the signals to be written to Seeq Server before any other work continues, you can
-            // call flushSignals() on the connection service.
-            this.connectionService.putSignal(signal);
-        }
+        // Loop through all the tags in our simulated datasource and tell Seeq Server about them
+        this.syncAssets(rootAssetId);
     }
 
     @Override
     public Stream<Sample> getSamples(GetSamplesParameters parameters) {
-        // Return a stream to iterate through all of the samples in the time range.
+        // This is an example of how you may query your datasource for tag values and is specific to the
+        // simulator example. This should be replaced with your own datasource-specific call.
+        Iterable<DatasourceSimulator.Tag.Value> tagValues = this.datasourceSimulator.getTagValues(
```

Comment:
I think it would be good to use an iterator or a stream. Your Iterable is actually a list. It would be hard to understand for an implementor why using stream down below would be beneficial if we use a list here.